### PR TITLE
Sanity check Python on OSX for LLDB tests

### LINKED ISSUE #32994: debuginfo-lldb tests crash in OS X 10.11 / Xcode 7.3
With a recent OS X/Xcode (here I'm using OS X 10.11.4 and Xcode 7.3), running the debuginfo-lldb tests results in a crash, even when using the system python install.

The crash pops up a window with an error: **Python quit unexpectedly while using the _lldb.so plug-in.**

```
---- [debuginfo-lldb] debuginfo-lldb/associated-types.rs stdout ----
	NOTE: compiletest thinks it is using LLDB version 350

error: Error while running LLDB
status: signal: 11
command: "/usr/bin/python2.7" "/Users/jturner/Source/borrowcksnip/rust/src/etc/lldb_batchmode.py" "x86_64-apple-darwin/test/debuginfo-lldb/associated-types.stage1-x86_64-apple-darwin" "x86_64-apple-darwin/test/debuginfo-lldb/associated-types.debugger.script"
stdout:
------------------------------------------
LLDB batch-mode script
----------------------
Debugger commands script is 'x86_64-apple-darwin/test/debuginfo-lldb/associated-types.debugger.script'.
Target executable is 'x86_64-apple-darwin/test/debuginfo-lldb/associated-types.stage1-x86_64-apple-darwin'.
Current working directory is '/Users/jturner/Source/borrowcksnip/rust'
Creating a target for 'x86_64-apple-darwin/test/debuginfo-lldb/associated-types.stage1-x86_64-apple-darwin'
settings set auto-confirm true

version
lldb-350.0.21.3 
command script import /Users/jturner/Source/borrowcksnip/rust/./src/etc/lldb_rust_formatters.py
type summary add --no-value --python-function lldb_rust_formatters.print_val -x ".*" --category Rust
type category enable Rust

breakpoint set --file 'associated-types.rs' --line 108
Breakpoint 1: where = associated-types.stage1-x86_64-apple-darwin`associated_types::assoc_struct::h4341f3f2e118615c + 27, address = 0x0000000100000a1b 
breakpoint set --file 'associated-types.rs' --line 115
Breakpoint 2: where = associated-types.stage1-x86_64-apple-darwin`associated_types::assoc_local::ha6fb01c9f512d9e5 + 52, address = 0x0000000100000ab4 
breakpoint set --file 'associated-types.rs' --line 119
Breakpoint 3: where = associated-types.stage1-x86_64-apple-darwin`associated_types::assoc_arg::h539559d4f7c8d367 + 16, address = 0x0000000100000ad0 
breakpoint set --file 'associated-types.rs' --line 127
Breakpoint 4: where = associated-types.stage1-x86_64-apple-darwin`associated_types::assoc_tuple::hf722967e21a7aab9 + 27, address = 0x0000000100000b2b 
breakpoint set --file 'associated-types.rs' --line 134
Breakpoint 5: where = associated-types.stage1-x86_64-apple-darwin`associated_types::assoc_enum::hcb7af09daa207720 + 98, address = 0x0000000100000ba2 
breakpoint set --file 'associated-types.rs' --line 137
Breakpoint 6: where = associated-types.stage1-x86_64-apple-darwin`associated_types::assoc_enum::hcb7af09daa207720 + 125, address = 0x0000000100000bbd 
run
Hit breakpoint 1.1: where = associated-types.stage1-x86_64-apple-darwin`associated_types::assoc_struct::h4341f3f2e118615c + 27, address = 0x0000000100000a1b, resolved, hit count = 1 
Process 24928 stopped * thread #1: tid = 0xea91e, 0x0000000100000a1b associated-types.stage1-x86_64-apple-darwin`associated_types::assoc_struct::h4341f3f2e118615c + 27 at associated-types.rs:108, queue = 'com.apple.main-thread', stop reason = breakpoint 1.1 frame #0: 0x0000000100000a1b associated-types.stage1-x86_64-apple-darwin`associated_types::assoc_struct::h4341f3f2e118615c + 27 at associated-types.rs:108 105 } 106 107 fn assoc_struct<T: TraitWithAssocType>(arg: Struct<T>) { -> 108 zzz(); // #break 109 } 110 111 fn assoc_local<T: TraitWithAssocType>(x: T) { Process 24928 launched: '/Users/jturner/Source/borrowcksnip/rust/x86_64-apple-darwin/test/debuginfo-lldb/associated-types.stage1-x86_64-apple-darwin' (x86_64) 
print arg
continue
Hit breakpoint 2.1: where = associated-types.stage1-x86_64-apple-darwin`associated_types::assoc_local::ha6fb01c9f512d9e5 + 52, address = 0x0000000100000ab4, resolved, hit count = 1 
print inferred
print explicitly
continue
Hit breakpoint 3.1: where = associated-types.stage1-x86_64-apple-darwin`associated_types::assoc_arg::h539559d4f7c8d367 + 16, address = 0x0000000100000ad0, resolved, hit count = 1 
print arg
continue
Hit breakpoint 4.1: where = associated-types.stage1-x86_64-apple-darwin`associated_types::assoc_tuple::hf722967e21a7aab9 + 27, address = 0x0000000100000b2b, resolved, hit count = 1 
print arg
continue
Hit breakpoint 5.1: where = associated-types.stage1-x86_64-apple-darwin`associated_types::assoc_enum::hcb7af09daa207720 + 98, address = 0x0000000100000ba2, resolved, hit count = 1 
print a
print b
continue
Hit breakpoint 6.1: where = associated-types.stage1-x86_64-apple-darwin`associated_types::assoc_enum::hcb7af09daa207720 + 125, address = 0x0000000100000bbd, resolved, hit count = 1 
print a
print b
continue
quit


------------------------------------------
stderr:
------------------------------------------

------------------------------------------

thread '[debuginfo-lldb] debuginfo-lldb/associated-types.rs' panicked at 'explicit panic', /Users/jturner/Source/borrowcksnip/rust/src/compiletest/runtest.rs:1613
note: Run with `RUST_BACKTRACE=1` for a backtrace.


failures:
    [debuginfo-lldb] debuginfo-lldb/associated-types.rs
```

### PR DESCRIPTION
**Pull request by alexcrichton**
Two primary changes:
- Don't get past the configure stage if `python` isn't coming from `/usr/bin` 
- Call `debugger.Terminate()` to prevent segfaults on newer versions of LLDB.

Closes #32994

### CODE REVIEW

**Comment by alexcrichton** [comment]
Note that tests still won't pass on the most recent OSX version due to what I believe is https://github.com/rust-lang/rust/issues/33062.

File: src/bootstrap/build/sanity.rs [hunk 1]
```diff
@@ -49,12 +54,29 @@ pub fn check(build: &mut Build) {
...
+    if build.config.build.ends_with("apple-darwin") {
+        if !python.starts_with("/usr/bin") {
+            panic!("
+
+Python detected ({}) is not /usr/bin/python and is unlikely to be able
+to run lldb tests. If this Python is installed via Homebrew then you'll likely
+need to uninstall it entirely (or move it out of 'PATH') to get lldb tests
+working.
+
+", python.display());
```
Comment (michaelwoerister) [inline src/bootstrap/build/sanity.rs]: Does this mean one cannot even build Rust when another Python version is in PATH?

Comment (alexcrichton) [inline src/bootstrap/build/sanity.rs]: I unfortunately didn't get that far in testing. If I had Homebrew Python installed and `/usr/local/bin/python` in my PATH then I'd still get weird errors about a missing "six" module, but once `/usr/local/bin` was removed from PATH that error went away. 

Python would successfully report an import of the `lldb` module (or at least the interpreter would exit cleanly) but would print some other errors during the `./configure` step.

I can try to get a clearer picture of what's going on soon as well.

Comment (michaelwoerister) [inline src/bootstrap/build/sanity.rs]: Couldn't we just adapt compiletest to always use `/usr/bin/python` for running the LLDB scripts on OSX?

Comment (alexcrichton) [inline src/bootstrap/build/sanity.rs]: Ok, so to be clear scary errors are printed out if `/usr/local/bin` is in `PATH` even if `/usr/bin/python` is used:

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Applications/Xcode.app/Contents/SharedFrameworks/LLDB.framework/Resources/Python/lldb/__init__.py", line 98, in <module>
    import six
ImportError: No module named six
```

That being said the `lldb_batchmode.py` script seems to run just fine and continue despite this error.

Unfortunately I still can't get tests passing due to https://github.com/rust-lang/rust/issues/33062, but it seems like yes, we should just tell compiletest to use /usr/bin/python unconditionally.

**Comment by alexcrichton** [comment]
Ok, I've modified to let `./configure` always pass but we just force `/usr/bin/python` on OSX for LLDB tests only

File: configure [hunk 2]
```diff
@@ -819,6 +819,19 @@ then
...
 
+# LLDB tests on OSX require /usr/bin/python, not something like Homebrew's
+# /usr/local/bin/python. We're loading a compiled module for LLDB tests which is
+# only compatible with the system.
+case $CFG_BUILD in
+    *-apple-darwin)
+        CFG_LLDB_PYTHON=/usr/bin/python
+        ;;
+    *)
+        CFG_LLDB_PYTHOn=$CFG_PYTHON
```
Comment (michaelwoerister) [inline configure]: `CFG_LLDB_PYTHOn` <- the lowercase 'n' is probably a typo.

Comment (alexcrichton) [inline configure]: Indeed!

**Comment by michaelwoerister** [comment]
Looks good. Thanks Alex!

**Comment by michaelwoerister** [comment]
These errors don't seem related...

**Pull request merged**